pkg/ingestor/parser/parser.go

```diff
@@ -113,18 +113,30 @@ func Subscribe(ctx context.Context, transportFunc func([]assembler.Graph) error)
 }
 
 // ParseDocumentTree takes the DocumentTree and create graph inputs (nodes and edges) per document node.
-func ParseDocumentTree(ctx context.Context, docTree processor.DocumentTree) ([]assembler.Graph, error) {
+func ParseDocumentTree(ctx context.Context, docTree processor.DocumentTree) ([]assembler.Graph, []*common.IdentifierStrings, error) {
```

Comment:
I firmly believe that having more than a tuple in return for a function makes it harder to maintain. I suggest we change this into a struct.

Reply:
Thanks for the review. I believe 2 values besides an error is reasonable. Will get input from comments from other maintainers as well.